meeting.js

```diff
@@ -0,0 +1,346 @@
+  document.getElementById("joinMeetingButton").addEventListener("click", joinMeeting);
+  document.getElementById("Video").addEventListener("click", toggleVideo);
+  document.getElementById("Microphone").addEventListener("click", toggleMicrophone);
+  document.getElementById("Microphone").innerHTML = '<img src="./images/mute.png" alt="Mute Icon"> Unmute';
+});
+
+
+
+// JOIN MEETING FUNCTIONALITY
+async function joinMeeting() {
+  document.getElementById("joinMeetingButton").style.display = "none";
+  $('#joinMeetingModal').modal('show');
+  var element = document.querySelector('.container3');
+  element.classList.add('blur');
+  try {
+    const meeting = await webex.meetings.create(sip);
+    currentMeeting = meeting;
+    setMediaListeners(meeting);
+
+    // Load the local video stream before joining the meeting
+    await loadCamera({ video: true, audio: false });
+
+    await meeting.join();
+    
+    // Initialize the video state based on the sendVideo parameter
+    isVideoStarted = true;
+   // const isSharing = meeting.getShareStatus();
+
+    // Add local video (and audio, if needed) to the meeting
+    await meeting.addMedia({
+      localStream: localMedia.cameraStream,
+      mediaSettings: {
+        receiveVideo:true,
+        receiveAudio: true,
+        receiveShare: false,
+        sendVideo: isVideoStarted,
+        sendAudio: false,
+        sendShare: false
+      }
+    });
+    // Update the video button text based on the video state
+    document.getElementById('Video').innerHTML = isVideoStarted ? '<img src="./images/video-camera.png" style="width: 25px; ;height:25px">Stop' : '<img src="./images/stop.png" style="width: 25px;height:25px">Start';
+    document.getElementById("Microphone").innerHTML = '<img src="./images/mute.png" alt="Mute Icon"> Unmute';
+    // UI updates for successfully joining the meeting
+  } catch (error) {
+    console.error('Failed to join or create the meeting', error);
+    alert('Failed to join or create the meeting: ' + error.message);
+  }
+}
+
+// LOAD CAMERA FUNCTIONALITY
+async function loadCamera(constraints) {
+  try {
+    // Stop any existing camera stream
+    if (localMedia.cameraStream) {
+      localMedia.cameraStream.getTracks().forEach(track => track.stop());
+    }
+
+    // Attempt to get a new camera stream
+    const stream = await navigator.mediaDevices.getUserMedia(constraints);
+    localMedia.cameraStream = stream;
+    document.getElementById('localVideo').srcObject = localMedia.cameraStream;
+
+    // Set up an event listener for the video track to handle the camera being turned off
+    const videoTrack = localMedia.cameraStream.getVideoTracks()[0];
+    if (videoTrack) {
+      videoTrack.onended = () => {
+        console.log('The camera has been turned off or disconnected.');
+        isVideoStarted = false;
+        document.getElementById('Video').innerHTML = '<img src="./images/stop.png" style="width: 25px;height:25px">Start';

```

Comment:
The SDK doesn't use/emit track events anymore. Please refactor this to use streams.